app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -156,22 +161,15 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
                 this@WebRenderFragment.session.url.let { url ->
                     when (value) {
                         NavigationEvent.VAL_CHECKED -> {
-                            CustomTilesManager.getInstance(context!!).pinSite(context!!, url,
+                            // FIXME: OverlayVM -> ToolbarVM -> PinnedTileRepo
+                            context!!.serviceLocator.pinnedTileRepo.addPinnedTile(url,
```

Comment:
In this case, we talk to the repo directly but below we talk through the view model: we should pick one and be consistent.

Reply:
As the comment suggests, this particular use case shouldn't be handled by `PinnedTileVM`. It should be handled by `ToolbarVM`. For now, we are just calling it directly to Repo (not a permanent solution). We can discuss about this in the upcoming meeting